module/x/peggy/spec/04_messages.md

```diff
@@ -82,83 +65,59 @@ This message will fail if:
 - Failure to build a batch of transactions.
 - If the orchestrator address is not present in the validator set
 
-### MsgConfirmBatch
-
-When a `MsgRequestBatch` is observed, validators need to sign batch request to signify this is not a maliciously created batch and to avoid getting slashed. 
+### MsgSubmitConfirm
 
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L137-143
-
-This message will fail if:
+When the peggy daemon witnesses an event on the counter chain the validator will submit confirmation of the event. 
 
-- The batch does not exist
-- If checkpoint generation fails
-- If a none validator address or delegated address 
-- If the counter chain address is empty or incorrect.
-- If counter chain address fails signature validation
-- If the signature was already presented in a previous message
+<!-- +++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L79-84 -->
 
-### MsgConfirmLogicCall
+```proto
+  message MsgSubmitClaim {
+    google.protobuf.Any confirm = 1 [
+        (cosmos_proto.accepts_interface) = "EthereumClaim"];
+    string signer = 2;
+  }
+```
 
-When a logic call request has been made, it needs to be confirmed by the bridge validators. Each validator has to submit a confirmation of the logic call being executed.
+There are three types of confirmations a validator can submit.
 
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L155-161
+- `CONFIRM_TYPE_VALSET`
+  - When the peggy daemon witnesses a complete validator set within the peggy module, the validator submits a signature of a message containing the entire validator set. 
```

Comment:
> peggy daemon witnesses a complete validator set within the peggy module

what does "witnessing a complete validator set" mean here? can you add more details?